src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java

```diff
@@ -32,58 +35,69 @@ public TrustCorporateValidator(AddressDtoValidator addressDtoValidator) {
-                            trustCorporateDto.getServiceAddressSameAsPrincipalAddress(), errors,
-                            loggingContext);
-
-                    if (isSameAddressFlagValid && Boolean.FALSE
-                            .equals(trustCorporateDto.getServiceAddressSameAsPrincipalAddress())) {
-                        validateAddress(TrustCorporateDto.SERVICE_ADDRESS_FIELD,
-                                trustCorporateDto.getServiceAddress(), errors, loggingContext);
-                    }
-
-                    validateIdentificationLegalForm(trustCorporateDto.getIdentificationLegalForm(), errors,
-                            loggingContext);
-                    validateIdentificationLegalAuthority(trustCorporateDto.getIdentificationLegalAuthority(), errors,
-                            loggingContext);
-
-                    boolean isOnRegisterInCountryFormedInIsValid =
-                            validateOnRegisteredInCountryFormedIn(trustCorporateDto.
-                                    getOnRegisterInCountryFormedIn(), errors, loggingContext);
-
-                    if (isOnRegisterInCountryFormedInIsValid && Boolean.FALSE
-                            .equals(trustCorporateDto.getOnRegisterInCountryFormedIn())) {
-                        validateOnRegisteredInCountryFormedInNotSupplied(trustCorporateDto, errors, loggingContext);
-                    } else if(isOnRegisterInCountryFormedInIsValid && Boolean.TRUE
-                            .equals(trustCorporateDto.getOnRegisterInCountryFormedIn())) {
-                        validateOnRegisteredInCountryFormedInSupplied(trustCorporateDto, errors, loggingContext);
-                    }
+                    validateTrustCorporateDto(trustCorporateDto, errors, loggingContext);
                 }
             }
         }
-
         return errors;
     }
 
+    private void validateTrustCorporateDto(TrustCorporateDto trustCorporateDto, Errors errors,
+            String loggingContext) {
+
+        validateName(trustCorporateDto.getName(), errors, loggingContext);
+
+        String type = trustCorporateDto.getType();
+        if (validateType(type, errors, loggingContext) && BeneficialOwnerType
+                .findByBeneficialOwnerTypeString(type)
+                .equals(BeneficialOwnerType.INTERESTED_PERSON)) {
+
+            validateDateBecameInterestedPerson(
+                    trustCorporateDto.getDateBecameInterestedPerson(),
+                    errors,
+                    loggingContext);
+        }
+
+        validateAddress(TrustCorporateDto.REGISTERED_OFFICE_ADDRESS_FIELD,
+                trustCorporateDto.getRegisteredOfficeAddress(), errors, loggingContext);
+
+        boolean isSameAddressFlagValid = validateServiceAddressSameAsPrincipalAddress(
+                trustCorporateDto.getServiceAddressSameAsPrincipalAddress(), errors,
+                loggingContext);
+
+        if (isSameAddressFlagValid && Boolean.FALSE
```

Comment:
Would `Boolean.FALSE.equals(trustCorporateDto.getServiceAddressSameAsPrincipalAddress())` be better as `!trustCorporateDto.getServiceAddressSameAsPrincipalAddress()`? If it returns (object) Boolean then ignore this comment!

Reply:
Honestly, I'm not sure I just copied this into another method to reduce the cognitive complexity.

It does return an instance of Boolean which may be null though I think `validateServiceAddressSameAsPrincipalAddress` checks for that.

Either way I'd say it's probably best to leave it

Reply:
I think if it's Boolean SonarQ wants Boolean.False.equals so I'd leave as is.